lib/rouge/lexers/xquery.rb

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    load_lexer 'xpath.rb'
+    class XQuery < XPath
+      title 'XQuery'
+      desc 'XQuery 3.1: An XML Query Language'
```

Comment:
Rouge allows for the delegation to other lexers as a means of allowing for multiple languages to be parsed. Does `XQuery` source text include XML/HTML syntax? Is this intended to be parsed?

Reply:
It does allow for XML-like syntax through [node constructors](https://www.w3.org/TR/xquery-31/#id-constructors), but also allows for XQuery code to be mixed in arbitrarily with `{ ... }`. For instance:

```xquery
for $item in /catalog return
<li id="{item/id}">{item/name}</li>
```

The `{item/id}` is especially problematic, as the XML lexer expects a full piece of valid XML. While `<li id="` might be able to be parsed by the XML lexer, `">` would not be managed nicely, as the XML lexer would start over from the root state to lex that. Because of that, I didn't see a viable way of delegating to the XML lexer.

Seeing that XQuery language specification documents how to parse these literal XML constructors quite well, and that the XML grammar is relatively simple, I saw it as a worthwhile trade-off to somewhat duplicate the XML parsing logic.